src/components/Button/Button.test.tsx

```diff
@@ -0,0 +1,122 @@
+import { fireEvent, render, screen } from '@testing-library/react'
+
+import { ButtonHierarchies, ButtonIconPositions, ButtonShapes, ButtonSizes, ButtonTypes } from './Button.types'
+import { Button } from '.'
+import { Icon } from '../Icon'
+
+const { Neutral, Danger } = ButtonHierarchies
+const { Right } = ButtonIconPositions
+const { Circle } = ButtonShapes
+const { Small, Large } = ButtonSizes
+const { Outline, Ghost } = ButtonTypes
+
+const icon = <Icon color="white" name="PiCircleHalfTiltLight" size={16} />
+
+describe('Button Component', () => {
+  beforeEach(() => {
+    jest.resetAllMocks()
+  })
+
+  test('renders primary filled button correctly', () => {
+    const { asFragment } = render(<Button>{'Button'}</Button>)
```

Comment:
using a container prop don't we avoid the fragment appearing in the snapshots? I would use it just in case

Reply:
It renders a `div` instead of a `DocumentFragment`, so it is a little useless